README.md

```diff
@@ -3,9 +3,16 @@
 ## Overview
 
 CharmHelpers provides an opinionated set of tools for building Juju
-charms that work together. In addition to basic tasks like interact-
-ing with the charm environment and the machine it runs on, it also
-helps keep you build hooks and establish relations effortlessly.
+charms that work together. In addition to basic tasks like interacting
+with the charm environment and the machine it runs on, it also helps
+keep you build hooks and establish relations effortlessly.
+
+## Why Python?
+
+* Low entry barrier, fairly common in automation tools;
+* An interpreted language helps with charm portability across different CPU architectures;
+* Doesn't require debugging symbols (just use pdb in-place);
+* An author or a user is able to make debugging changes without recompiling a charm.
```

Comment:
I would drop the first bullet point, and probably replace it with:

* Python is an extremely popular, easy to learn, and powerful language which is also common in automation tools.